src/LinearOperators.jl

```diff
@@ -470,6 +469,29 @@ check_positive_definite(M :: AbstractMatrix) = check_positive_definite(LinearOpe
 
 # Special linear operators.
 
+"""`opEye()`
+
+Identity operator.
+```
+opI = opEye()
+v = rand(5)
+@assert opI * v === v
+```
+"""
+struct opEye <: AbstractLinearOperator{Any,Nothing,Nothing,Nothing} end
+
+*(::opEye, x :: AbstractArray{T,1} where T) = x
+*(x :: AbstractArray{T,1} where T, ::opEye) = x
+*(::opEye, A :: AbstractArray{T,2} where T) = A
+*(A :: AbstractArray{T,2} where T, ::opEye) = A
+*(::opEye, T :: AbstractLinearOperator) = T
+*(T :: AbstractLinearOperator, ::opEye) = T
+*(::opEye, ::opEye) = opEye()
```

Comment:
Out of curiosity, why doesn't this last operation return one of the two arguments? By convention, I would return the one on the right.

Reply:
You're right. Since this is a singleton I didn't think about that, but it's probably cheaper.